Create an `OrnamentButton`

JPKribs opened the pull request:
### Summary

Resolves: https://github.com/jellyfin/Swiftfin/issues/1532

Breaking this out of another PR. Trying to do a better job splitting out items that can be standalone PRs. This is a simple button that tries to mirror's Apples newer VisionOS Ornaments style: https://developer.apple.com/design/human-interface-guidelines/ornaments

I think this roughly mirrors what Apple is using with the exception of the disabled version. I think apple just hides these when disabled but I don't have a real world example of this. Also, Apple doesn't have a shadow which I added since it gives it a layering/popping effect. Open to discussion

My thinking this would replace our Close Button and Menu Ellipsis. Down the road, it looks like this is where Apple puts their download button which I think looks nice.

Totally open to feedback! If we hate this, I'm also game to close it out. I just thought Apple's frosted buttons they've been using lately looked solid. I was going to use something like this in [PR #1541](https://github.com/jellyfin/Swiftfin/pull/1541) but I thought it would be better to put this in it's own PR so this got some discussion on it's own merit first.

---

### Screenshots

**Light Mode**

<img width="163" alt="Screenshot 2025-05-21 at 00 23 22" src="https://github.com/user-attachments/assets/1d6d42c8-9b90-449c-8a66-5e31c1399845" />

**Dark Mode**

<img width="168" alt="Screenshot 2025-05-21 at 00 23 49" src="https://github.com/user-attachments/assets/c8a1956c-03e6-4008-8d25-b4f62d05d026" />

### Videos

**Light Mode**

https://github.com/user-attachments/assets/1e15e1ae-33c4-4b83-b149-1283ce1bd429

**Dark Mode**

https://github.com/user-attachments/assets/ecf76f08-ee41-40ae-8556-09fe7c50bc54

### Apple Examples

**Music**

![image](https://github.com/user-attachments/assets/e0761a83-d00e-4a8d-88f2-029e8c24c979)

**Podcasts**

<img width="332" alt="Screenshot 2025-05-20 at 22 41 57" src="https://github.com/user-attachments/assets/f97e90a6-0f1a-4d8c-ac7b-12bb031b9e3f" />

thornbill commented:
The shadow doesn't seem to match with Apple's design.

JPKribs commented:
> The shadow doesn't seem to match with Apple's design.

Good call! I've cleaned that up in this PR and updated this in the main post.

LePips commented:
I do really like this attention to detail and can do a closer audit later, but I would like it implemented as a `ButtonStyle` instead of a custom button entirely.

JPKribs commented:
Good call. I've moved that over. This likely pushes this to after  iOS 16 since we would need the [ButtonMenuStyle](https://developer.apple.com/documentation/swiftui/buttonmenustyle) for the ellipsis menu. 

This is all in my latest commit and I think it's in a good spot but I have this:

```
                    if #available(iOS 16.0, *) {
                        Menu(L10n.options, systemImage: "ellipsis") {
                            items()
                        }
                        .menuStyle(.button)
                        .buttonStyle(.ornament)
                    } else {
                        Menu(L10n.options, systemImage: "ellipsis.circle") {
                            items()
                        }
                        .labelStyle(.iconOnly)
                        .backport
                        .fontWeight(.semibold)
                        .foregroundStyle(accentColor)
                    }
```

LePips commented:
Apologies for not commenting on this earlier. I do really like this, but I would like to find a way to unify the design of our buttons alongside the work in #1547, and porting it to existing buttons.

Also, we would still need this stuff even with iOS 26 releasing. Frankly, we will most likely opt out of the liquid glass design until we support iOS 26 as a minimum, but I have no guarantees since I haven't played with it yet.

https://www.donnywals.com/opting-your-app-out-of-the-liquid-glass-redesign-with-xcode-26/

JPKribs commented:
> Apologies for not commenting on this earlier. I do really like this, but I would like to find a way to unify the design of our buttons alongside the work in #1547, and porting it to existing buttons.
> 
> Also, we would still need this stuff even with iOS 26 releasing. Frankly, we will most likely opt out of the liquid glass design until we support iOS 26 as a minimum, but I have no guarantees since I haven't played with it yet.
> 
> https://www.donnywals.com/opting-your-app-out-of-the-liquid-glass-redesign-with-xcode-26/

No worries at all! I'll keep the branch alive if we want to pull from it but my thinking was this style was pretty short lived so it doesn't make a ton of sense to migrate to a new custom button if there is an Apple-native version on the horizon.

LePips commented:
With #1547, I would like to see these also as the `.material/tintedMaterial` button styles. However, now we will need to edit the button style to adhere to a container shape instead of assuming a rounded rectangle. There is a `TODO` documenting that.

JPKribs commented:
> I would like to see these also as the .material/tintedMaterial button styles.

I was thinking the same thing when I saw it. I'll take a look at this this afternoon

JPKribs commented:
Here is what has changed:

Migrate over to `Material` for`PillHStack`.
Migrate Close & Menu buttons in Navigation with `TintedMaterial`
Migrate filters over to `TintedMaterial` & update `TintedMaterial` to have an optional `isPressed`
Remove `isPressed` since the only usages are in `TintedMaterial`
Remove the static rounding on `TintedMaterial` to instead go in a case by case bases so I can do circles for some and rounded rectangles for others.

## Results:

### ItemViews

For `ItemViews`, the menu ellipsis and close buttons are new. Along with the Genre/Studio buttons are now using the `material` styling as well.

**Light:**
https://github.com/user-attachments/assets/4ff05d91-498c-40ab-ad02-2be6e9db710b

**Dark:**
https://github.com/user-attachments/assets/412dcdd7-89f4-457a-8cd7-d90ee54f5a90

### Active Session & Server Activity

Both of these locations had `isPressed` logic and `line.3.horizontal.decrease.circle`. I changed these to use `line.3.horizontal.decrease` and `Material` instead. Additionally, I removed the `isPressed` button style and replaced it with an optional `isPressed` option for `Material` since these were only used on buttons that are now `Material` 

https://github.com/user-attachments/assets/1e4eaa75-ffb1-4598-947b-f2ed5be1be44
